nbcelltests/tests/test_test.py

```diff
@@ -5,9 +5,223 @@
 # This file is part of the nbcelltests library, distributed under the terms of
 # the Apache License 2.0.  The full license can be found in the LICENSE file.
 #
-# for Coverage
+import tempfile
+import os
+import sys
+import unittest
 
+from nbcelltests.test import run
 
-class TestTests:
-    def test_run(self):
+# TODO: we should generate the notebooks rather than having them as files
+# (same for lint ones)
+CUMULATIVE_RUN = os.path.join(os.path.dirname(__file__), '_cumulative_run.ipynb')
+CELL_ERROR = os.path.join(os.path.dirname(__file__), '_cell_error.ipynb')
+TEST_ERROR = os.path.join(os.path.dirname(__file__), '_test_error.ipynb')
+TEST_FAIL = os.path.join(os.path.dirname(__file__), '_test_fail.ipynb')
+
+# Hack. We want to test expected behavior in distributed situation,
+# which we are doing via pytest --forked.
+FORKED = '--forked' in sys.argv
+
+
+def _assert_x_undefined(t):
+    t.run_test("""
+    try:
+        x
+    except NameError:
         pass
+    else:
+        raise Exception('x was already defined')
+    """)
+
+# TODO: This test file's manual use of unittest is brittle
+
+
+def _import_from_path(pypath, f):
+    import importlib.util
+    spec = importlib.util.spec_from_file_location(pypath, f)
+    mod = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(mod)
+    return mod
```

Comment:
While I appreciate that this is just our testing code, given my previous experiences with importlib, I would strongly prefer this to be tested on the full 3.5 - 3.8 set of Python versions. Not a blocker though, just a heads up that there might be some differences across versions.

Reply:
Ah, there was a TODO comment to exactly this effect. I wonder where it went. I'll restore it!

Reply:
> that this is just our testing code

What kind of attitude is that in a library about testing code, ha ha 😲

Reply:
It's like how I wanted to turn off the linter while I was working on the linting part.